### ISSUE TITLE
Microsoft.Extensions.DependencyInjection.Abstractions: Unable to resolve service for type 

### ISSUE THREAD

**Issue by exhesham** [Azure/azure-functions-dotnet-extensions#14]
I am receiving the next error when running the `Basic` Demo:
```
[5/23/2019 12:38:21 PM] Executed 'SampleFunction' (Failed, Id=d27a090a-4ddf-4e5c-92e6-660646c81be9)
[5/23/2019 12:38:21 PM] Microsoft.Extensions.DependencyInjection.Abstractions: Unable to resolve service for type 'Microsoft.Azure.Functions.Samples.DependencyInjectionBasic.IGreeter' while attempting to activate 'Microsoft.Azure.Functions.Samples.DependencyInjectionBasic.SampleFunction'.
[5/23/2019 12:38:22 PM] An unhandled host error has occurred.
[5/23/2019 12:38:22 PM] Microsoft.Extensions.DependencyInjection.Abstractions: Unable to resolve service for type 'Microsoft.Azure.Functions.Samples.DependencyInjectionBasic.IGreeter' while attempting to activate 'Microsoft.Azure.Functions.Samples.DependencyInjectionBasic.SampleFunction'.

```

What am I missing?

**Comment by Ohmniox** [comment]
Facing the same issue, Any help?

**Comment by ErikAndreas** [comment]
Me too: Microsoft.Extensions.DependencyInjection.Abstractions: Unable to resolve service for type [redacted] while attempting to activate ...

Any input?
Azure Functions Core Tools (2.7.1633 Commit hash: 45c7d86a3bbc9ed0a80a8f4199aa7ea80ccfb24e)
Function Runtime Version: 2.0.12673.0

**Comment by ErikAndreas** [comment]
@fabiocav is this same issue as e.g https://github.com/Azure/azure-functions-host/issues/4939, https://github.com/Azure/azure-functions-host/issues/4914 and or https://github.com/Azure/azure-functions-host/issues/3399? A bit hard to follow and what's really related (type of DI and or service lifetime) as well as what is fixed (by 1.0.30)?

AddHttpClient and AddDbContext (EF) seems to work just fine

**Comment by exhesham** [comment]
I moved to AWS Lambda due to too much problems with Azure.
